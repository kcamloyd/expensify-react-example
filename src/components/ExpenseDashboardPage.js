import React from 'react'
import ConnectedExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ConnectedExpensesSummary from './ExpensesSummary'

const ExpenseDashboardPage = () => (
    <div>
        <ConnectedExpensesSummary />
        <ExpenseListFilters />
        <ConnectedExpenseList />
    </div>
)

export default ExpenseDashboardPage