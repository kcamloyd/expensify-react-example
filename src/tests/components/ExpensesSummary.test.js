import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should render ExpensesSummary with one expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with multiple expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={14} expensesTotal={223945} />)
    expect(wrapper).toMatchSnapshot()
})