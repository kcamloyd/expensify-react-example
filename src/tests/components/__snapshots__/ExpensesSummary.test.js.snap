// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with multiple expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
    14
     
    expenses
     totaling 
    $2,239.45
  </h1>
</div>
`;

exports[`should render ExpensesSummary with one expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;
